samples/apps/logging/logging.cpp

```diff
@@ -878,7 +904,26 @@ namespace loggingapp
 
             // SNIPPET_START: user_data_check
             // Check caller's user-data for required permissions
-            const nlohmann::json user_data = caller_ident.user_data;
+            nlohmann::json user_data = nullptr;
+            auto result =
+              get_user_data_v1(ctx.tx, caller_ident.user_id, user_data);
+            if (result == ccf::ApiResult::NotFound)
+            {
+              return ccf::make_error(
+                HTTP_STATUS_NOT_FOUND,
+                ccf::errors::ResourceNotFound,
+                fmt::format("User {} does not exist", caller_ident.user_id));
```

Comment:
I don't think this message is correct? Its possible to be a real, known user, but have no user data, in which case we'd get a `NotFound`?